### PR TITLE
guide: Add Dart Edge guide

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

[Dart Edge](https://docs.dartedge.dev/) is a project created by [Invertase](https://invertase.io/), our partner, and it allows developers write edge functions in Dart programming language, so Flutter developers can keep using their favorite programing language for writing their backend.

This PR adds a guide on Dart Edge within the functions guide sections.

### CODE REVIEW

**Comment by dshukertjr** [comment]
@thorwebdev I wanted to add a guide on Dart Edge, but wasn't exactly sure where to put it. I have it under the `examples` section within the functions docs, but do you think this is fine?

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks

File: apps/docs/pages/guides/functions/examples/dart-edge.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+export const meta = {
+  id: 'examples-dart-edge',
+  title: 'Dart Edge',
+  description: 'Write your functions using Dart.',
+}
+
+[Dart Edge](https://docs.dartedge.dev/) is a project that enables Dart code to be run on different edge environments built by [Invertase](https://invertase.io/). This guide will walk you through how you can start using Dart Edge to write your backend code using Dart and run it on Supabase.
+
+> ⚠️ WARNING: Dart Edge is an experimental project - it's probably not ready for production usage unless you're okay with living on the 'edge'.
```
Comment (laktek) [inline apps/docs/pages/guides/functions/examples/dart-edge.mdx:11]: Can we please add a link here on who to contact on support issues for this?

**Comment by supabase-enjoyer** [comment]
OMG GUYS YOU ARE AWESOME !!!!!!!!!!!

**Pull request merged**